test/performance/latency/LatencyTestPublisher.cpp

```diff
@@ -820,12 +915,28 @@ void LatencyTestPublisher::print_stats(
 void LatencyTestPublisher::export_raw_data(
         uint32_t datasize)
 {
-    std::ofstream data_file;
-    data_file.open(raw_data_file_, std::fstream::app);
-    for (std::vector<std::chrono::duration<double, std::micro> >::iterator tit = times_.begin(); tit != times_.end();
+    ofstream data_file;
+    data_file.open(raw_data_file_, fstream::app);
+    for (vector<chrono::duration<double, micro>>::iterator tit = times_.begin(); tit != times_.end();
             ++tit)
     {
-        data_file << ++raw_sample_count_ << "," << datasize << "," << (*tit).count() << std::endl;
+        data_file << ++raw_sample_count_ << "," << datasize << "," << (*tit).count() << endl;
     }
     data_file.close();
 }
+
+int32_t LatencyTestPublisher::total_matches() const
+{
+    // no need to lock because is used always within a
+    // condition variable wait predicate
+
+    int32_t count = data_writer_listener_.matched_
+            + data_reader_listener_.matched_
+            + command_writer_listener_.matched_
+            + command_reader_listener_.matched_;
+
+    // Each endpoint has a mirror counterpart in the LatencyTestPublisher
+    // thus, the maximun number of matches is 4
+    assert(count >= 0 && count < 5);
```

Comment:
This assumption is false. The maximum is `4 * subscribers_`

Reply:
Done in 38d969f4b12f32e97ec60f8b44d02e591713638e